metals/src/main/scala/scala/meta/internal/metals/debug/DebugProxy.scala

```diff
@@ -60,9 +61,27 @@ private[debug] final class DebugProxy(
       outputTerminated = true
       server.send(request)
     case request @ SetBreakpointRequest(args) =>
-      handleSetBreakpointsRequest(args)
-        .map(DebugProtocol.syntheticResponse(request, _))
-        .foreach(client.consume)
+      import scala.meta.internal.metals.MetalsEnrichments._
+      val path = adapters.adaptPath(args.getSource.getPath).toAbsolutePath
+      if (sourcePathProvider.isWorkspacePath(path)) {
+        server
+          .sendPartitioned(List(request))
+          .map(_.map(DebugProtocol.parseResponse[SetBreakpointsResponse]))
+          .map(_.head.get)
+          .map { response =>
+            val breakpoints = response.getBreakpoints
+            if (breakpoints != null) {
+              breakpoints.foreach(_.setSource(args.getSource))
```

Comment:
What are we doing here?

Reply:
java-debug is not setting the sources for breakpoints, so we have to do it manually :(

Reply:
Hmm, why is `java-debug` setting the sources for breakpoints in the former case and not now?

Reply:
By former case you mean the library sources? I have to set them there as well: https://github.com/scalameta/metals/blob/master/metals/src/main/scala/scala/meta/internal/metals/debug/SetBreakpointsRequestHandler.scala#L50